nuxeo-ai-core/src/main/java/org/nuxeo/ai/enrichment/EnrichmentProvider.java

```diff
@@ -56,19 +61,19 @@ public interface EnrichmentProvider {
      */
     @SuppressWarnings("unchecked")
     default RetryPolicy getRetryPolicy() {
-        return new RetryPolicy().abortOn(NuxeoException.class, FatalEnrichmentError.class)
-                                .withMaxRetries(2)
-                                .withBackoff(3, 36, TimeUnit.SECONDS);
+        return new RetryPolicy<Collection<AIMetadata>>()
+            .abortOn(NuxeoException.class, FatalEnrichmentError.class)
+            .withBackoff(3,36, ChronoUnit.SECONDS)
+            .withMaxRetries(2);
     }
 
     /**
      * The circuit breaker for the provider
      */
     default CircuitBreaker getCircuitBreaker() {
```

Comment:
Same thing than getRetryPolicy:
```suggestion
    default CircuitBreaker<Collection<AIMetadata>> getCircuitBreaker() {
```